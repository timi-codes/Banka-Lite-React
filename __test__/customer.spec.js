import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter, Switch } from 'react-router-dom';
import CustomerDashboard from '@pages/customer'
import { Provider } from 'react-redux';
import {
  render,
  cleanup,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

const mockStore = configureMockStore([thunk]);
let store;
const initialSate = {
  isPending: false,
  isAuthenticated: false,
  error: null,
  user:{},
  accounts: []
};

const props = {
  isPending: false,
  error: null,
  accounts: [
    {
      
    }
  ],
   user: {
     email: 'email@gmail.com'
   },
  getAccounts: jest.fn(),
  history: {
    push: jest.fn()
  },
  getCurrentUser: jest.fn()
};

const renderWithRTL = state => {
  store = mockStore({
    auth: {
      isPending: false,
      error: null,
      isAuthenticated: true,
      user: {
        email: 'email@gmail.com'
      },
    },
    account: {
      ...state
    }
  });
  return {
    ...render(
      <Provider store={store}>
        <BrowserRouter>
          <Switch>
            <CustomerDashboard {...props} />
          </Switch>
        </BrowserRouter>
      </Provider>
    ),
    store
  };
};

describe('Test signup page', () => {
  afterEach(cleanup);
  test('should render customer dashboard when empty', async () => {
    const { container } = renderWithRTL(initialSate);
    expect(container).toMatchSnapshot()

  });

  test('should render customer table with account', async () => {
    const { container } = renderWithRTL({
      accounts:[
        {
          accountNumber: "10184349343403",
          firstName: "Timi",
          lastName: "Tejumola",
          type: "savings",
          balance: "1000",
          status: "dormant",
          createdOn: new Date()
        }
      ]
    });
    expect(container).toMatchSnapshot()

  });
});