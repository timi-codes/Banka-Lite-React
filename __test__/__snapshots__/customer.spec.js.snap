// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test signup page should render customer dashboard when empty 1`] = `
<div>
  <div
    class="dash-wrapper"
  >
    <header>
      <div
        class="top"
      >
        <img
          alt="logo"
          id="logo"
          src="icon-mock"
        />
        <div
          class="current-user web-user dropdown"
        >
          <img
            alt="Avatar"
            class="avatar"
            src="[object Object]"
          />
          <h4
            id="current-user"
          >
            email@gmail.com
          </h4>
          <img
            alt="Arrow Down"
            class="arrow-down"
            src="icon-mock"
          />
          <div
            class="dropdown-content"
          >
            <a
              href="/"
            >
              Logout
            </a>
          </div>
        </div>
      </div>
    </header>
    <div
      class="banner"
    >
      <div
        class="balance"
      >
        <p>
          Account Balance
        </p>
        <h3
          id="current-balance"
        >
          0.00
        </h3>
      </div>
      <button
        id="account-button"
        type="button"
      >
        Create an account
      </button>
    </div>
    <div
      class="content"
    >
      <div
        id="accountText"
      >
        No account has been created
      </div>
    </div>
  </div>
</div>
`;

exports[`Test signup page should render customer table with account 1`] = `
<div>
  <div
    class="dash-wrapper"
  >
    <header>
      <div
        class="top"
      >
        <img
          alt="logo"
          id="logo"
          src="icon-mock"
        />
        <div
          class="current-user web-user dropdown"
        >
          <img
            alt="Avatar"
            class="avatar"
            src="[object Object]"
          />
          <h4
            id="current-user"
          >
            email@gmail.com
          </h4>
          <img
            alt="Arrow Down"
            class="arrow-down"
            src="icon-mock"
          />
          <div
            class="dropdown-content"
          >
            <a
              href="/"
            >
              Logout
            </a>
          </div>
        </div>
      </div>
    </header>
    <div
      class="banner"
    >
      <div
        class="balance"
      >
        <p>
          Account Balance
        </p>
        <h3
          id="current-balance"
        >
          0.00
        </h3>
      </div>
      <button
        id="account-button"
        type="button"
      >
        Create an account
      </button>
    </div>
    <div
      class="content"
    >
      <div
        class="css-0"
      >
        <div
          class="css-ibvn5k"
        />
        <div
          class="css-18h190t"
        />
        <div
          class="css-1bt58ph"
        />
      </div>
      <div
        class="tableDiv"
      >
        <table
          id="account-table"
        >
          <thead>
            <tr>
              <th
                class="table-title"
                colspan="6"
              >
                My Accounts
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <th
                class="table-header"
              />
              <th
                class="table-header"
              >
                Type
              </th>
              <th
                class="table-header"
              >
                Account Number
              </th>
              <th
                class="table-header"
              >
                Status
              </th>
              <th
                class="table-header"
              >
                Created On
              </th>
              <th
                class="table-header"
              >
                Balance
              </th>
            </tr>
            <tr>
              <td>
                <img
                  alt="savings"
                  src="icon-mock"
                />
              </td>
              <td>
                savings
              </td>
              <td>
                10184349343403
              </td>
              <td>
                dormant
              </td>
              <td>
                28, August 2019
              </td>
              <td>
                â‚¦
                1000
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
